students/eowyn/backcast.py

```diff
@@ -0,0 +1,252 @@
+import pandas as pd
+import numpy as np
+from tabulate import tabulate
+
+fname = 'M426_Processed_Data.txt'
+metdf = pd.read_table(fname, skiprows=1,
+                      index_col=0,
+                      parse_dates=True,
+                      infer_datetime_format=True)
+metdf.index.name = 'DateTime'
+
+"""
+Parse column names to determine sensor info: measurement height,
+sensor type, and field type (mean, max, etc.) handling the existance
+of fields like "St Dev" that need to be combined to "StDev" and
+retaining only the numeric component of the sensor height (in m or ft)
+"""
+colnames = list(metdf)
+sensorInfo = []
+for col in colnames:
+    sensor = col.split(' ')
+    unitIndex, conversion = sensor[0].lower().find('m'), 1.0
+    if unitIndex == -1:
+        unitIndex, conversion = sensor[0].lower().find('ft'), 0.3048
+    sensorInfo.append([float(sensor[0][:unitIndex]) * conversion,
+                       sensor[1],
+                       "".join(sensor[2:])])
+
+# Locate column name index of highest mean anemometer sensor
+windAve = [(item[2] == "Mean" and item[1] == "Anemometer")
+           for item in sensorInfo]  # Boolean where field is ave wind
+heights = [item[0] for item in sensorInfo]
+meanWindHeights = [a * b for a, b in zip(windAve, heights)]
+mxInd = meanWindHeights.index(max(meanWindHeights))
+
+# Drop all other columns from dataframe
+dropcols = colnames[:mxInd] + colnames[mxInd + 1:]
+metdf.drop(dropcols, axis=1, inplace=True)
+
+# Replace invalid data with nan
+metdf.iloc[:, 0] = metdf.iloc[:, 0].replace(-99.99, np.nan)
```

Comment:
I'm kind of surprised that's not something built in to Pandas somehow. But I guess not.